packages/next/next-server/server/config.ts

```diff
@@ -10,7 +10,60 @@ import * as Log from '../../build/output/log'
 const targets = ['server', 'serverless', 'experimental-serverless-trace']
 const reactModes = ['legacy', 'blocking', 'concurrent']
 
-const defaultConfig: { [key: string]: any } = {
+interface Config {
+  env: any[]
+  webpack: any | null
+  webpackDevMiddleware: any | null
+  distDir: string
+  assetPrefix: string
+  configOrigin: string
+  useFileSystemPublicRoutes: boolean
+  generateBuildId: () => null
+  generateEtags: boolean
+  pageExtensions: string[]
+  target: 'server' | 'serverless' | 'experimental-serverless-trace'
```

Comment:
```suggestion
  target: 'server' | 'serverless'
```

Let's not expose experiments.